src/frontend/http.rs

```diff
@@ -64,139 +74,276 @@ struct QueryBody {
+
+            // Load the file content from the request
+            // TODO: we're actually buffering the entire file into memory here which is sub-optimal,
+            // since we could be writing the contents of the stream out into a temp file on the disk
+            // to have a smaller memory footprint. However, for this to be supported warp first
+            // needs to support streaming here: https://github.com/seanmonstar/warp/issues/323
+            let value = p
+                .stream()
+                .try_fold(Vec::new(), |mut vec, data| {
+                    vec.put(data);
+                    async move { Ok(vec) }
+                })
+                .await
+                .map_err(|e| {
+                    eprintln!("Error reading part's data: {}", e);
+                    warp::reject::reject()
+                })
+                .unwrap();
+
+            // Parse the schema and load the file contents into a vector of record batches
+            let (schema, partition) = match filename.split('.').last().unwrap() {
+                "csv" => {
+                    let schema = match csv_schema.clone() {
+                        Some(schema) => schema,
+                        None => {
+                            // TODO: We could actually make the schema part optional, and utilize
+                            // CSV reader's ability to infer the schema, so that we gain some
+                            // ergonomics at the expense of some schema ambiguity
                             return warp::reply::with_status(
-                                "NOT_MODIFIED",
-                                StatusCode::NOT_MODIFIED,
+                                "CSV schema part not supplied".to_string(),
+                                StatusCode::BAD_REQUEST,
                             )
                             .into_response();
                         }
-                    }
+                    };
 
-                    // Guess we'll have to actually run the query
-                    let physical = context.create_physical_plan(&plan).await.unwrap();
-                    let batches = context.collect(physical).await.unwrap();
+                    let builder = ReaderBuilder::new()
+                        .with_schema(Arc::new(schema.clone()))
+                        .has_header(true)
+                        .with_escape(b'\\'); // default is None, change to \
 
-                    let mut buf = Vec::new();
-                    let mut writer = LineDelimitedWriter::new(&mut buf);
-                    writer.write_batches(&batches).unwrap();
-                    writer.finish().unwrap();
+                    let mut cursor = Cursor::new(&value);
+                    let csv_reader = builder.build(&mut cursor).unwrap();
+                    let partition: Vec<RecordBatch> =
+                        csv_reader.into_iter().map(|item| item.unwrap()).collect();
```

Comment:
this avoids the extra map (arguably it's more code, but it lets us postpone the unwrap() until the very end; good for when we actually start propagating errors upwards instead of panicking).

```suggestion
                        csv_reader.into_iter().collect::<Result<Vec<RecordBatch>, ArrowError>>().unwrap();
```